MiniF/users/serializers.py

```diff
@@ -128,6 +137,12 @@ def validate(self, attrs):
         if not user:
             raise serializers.ValidationError("Invalid credentials")
 
+        try:
+            logger.info(f"Triggering welcome email for logged in user: {user.email}")
+            send_welcome_email.delay(user.id)
```

Comment:
This is acceptable, but the DRF serializer should only return data, not handle asynchronous logic.

It's better to put this in the post_save signal:
    
    # users/signals.py
    @receiver(post_save, sender=UserProfile)
    def send_welcome_email_signal(sender, instance, created, **kwargs):
        if created:
            send_welcome_email.delay(instance.id)